### PR TITLE
Update cli commands to work with keystore.json files OR  with pem files

### PR DESCRIPTION
**Pull request by miiu96**

### CODE REVIEW

File: erdpy/wallet/keyfile.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    plaintext = decryptor.update(ciphertext) + decryptor.finalize()
+    pemified_private_key = b2a_base64(hexlify(plaintext))
+
+    hmac_key = key[16:32]
+    h = hmac.HMAC(hmac_key, hashes.SHA256(), backend=backend)
+    h.update(ciphertext)
+    mac = h.finalize()
+
+    if mac != unhexlify(keystore['crypto']['mac']):
+        print('invalid keystore (MAC does not match)')
```
Comment (andreibancioiu) [inline erdpy/wallet/keyfile.py:57]: Please raise `errors.SomeSpecificError()`.

File: erdpy/wallet/keyfile.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+def load_from_key_file(key_file_json, password):
+    with open(key_file_json) as json_f:
+        keystore = load(json_f)
+
+    backend = default_backend()
+
+    # derive the decryption key
+    kdf_name = keystore['crypto']['kdf']
+    if kdf_name != 'scrypt':
+        print('unknown key derivation function', kdf_name)
```
Comment (andreibancioiu) [inline erdpy/wallet/keyfile.py:24]: Please raise error.

File: erdpy/cli_contracts.py [hunk 3]
```diff
@@ -44,7 +47,11 @@ def setup_parser(subparsers):
     sub = subparsers.add_parser("call")
     sub.add_argument("contract")
     sub.add_argument("--proxy", required=True)
-    sub.add_argument("--pem", required=True)
+
+    sub.add_argument("--pem", required=not (is_arg_present("--keyfile", sys.argv)), help="the PEM file of the owner")
```
[the remark this answers is not in the document]
Comment (miiu96) [reply erdpy/cli_contracts.py:51]: done.

File: erdpy/cli_transactions.py [hunk 4]
```diff
@@ -31,7 +32,8 @@ def setup_parser(subparsers):
 
     sub = subparsers.add_parser("new", description="Create a new regular transaction")
     _add_common_arguments(sub)
-    sub.add_argument("--outfile", type=FileType("w"), default=sys.stdout, help="where to save the signed transaction, the hash")
+    sub.add_argument("--outfile", type=FileType("w"), default=sys.stdout, help="where to save the signed transaction, "
```
[the remark this answers is not in the document]
Comment (miiu96) [reply erdpy/cli_transactions.py:35]: there are no changes this .flake8 file

File: erdpy/wallet/test/test_decode.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import unittest
+
+from erdpy.wallet.keyfile import load_from_key_file, get_password
+
+
+class TestDecode(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (miiu96) [reply erdpy/wallet/test/test_decode.py:6]: done.

File: erdpy/wallet/test/test_decode.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+    def test_decode(self):
+        password = get_password("pass.txt")
+        address, seed = load_from_key_file("keyfile.json", password)
+
+        print(address)
+        print(seed)
+        print(password)
+
+        self.assertEqual("1", "1")
```
[the remark this answers is not in the document]
Comment (miiu96) [reply erdpy/wallet/test/test_decode.py:16]: done. test refactored

File: erdpy/wallet/keyfile.py (hunk 1 above)
Comment (miiu96) [reply erdpy/wallet/keyfile.py:57]: done.

File: erdpy/wallet/keyfile.py (hunk 2 above)
Comment (miiu96) [reply erdpy/wallet/keyfile.py:24]: done.

**Pull request merged**